contracts/sol6/Dao/EmergencyFeeHandler.sol

```diff
@@ -0,0 +1,201 @@
+pragma solidity 0.6.6;
+
+import "../IKyberFeeHandler.sol";
+import "../utils/PermissionGroupsNoModifiers.sol";
+import "../utils/zeppelin/ReentrancyGuard.sol";
+import "../utils/zeppelin/SafeMath.sol";
+import "../utils/Utils5.sol";
+
+contract EmergencyKyberFeeHandler is IKyberFeeHandler, PermissionGroupsNoModifiers, ReentrancyGuard, Utils5 {
+    using SafeMath for uint256;
+
+    uint16 public immutable rewardBps;
+    uint16 public immutable rebateBps;
+    address public kyberNetwork;
+
+    mapping(address => uint256) public feePerPlatformWallet;
+    uint256 public totalPlatformFeeWei; // total balance in the contract that is for platform fee
+    mapping(address => uint256) public rebatePerWallet;
+    uint256 public totalRewardWei;
+
+    event HandleFeeFailed(address[] rebateWallets, uint256[] rebateBpsPerWallet, uint256 feeBRRWei);
+
+    event FeeDistributed(
+        address indexed platformWallet,
+        uint256 platformFeeWei,
+        uint256 rewardWei,
+        uint256 rebateWei,
+        address[] rebateWallets,
+        uint256[] rebatePercentBpsPerWallet,
+        uint256 burnAmtWei
+    );
+
+    event EtherWithdraw(uint256 amount, address sendTo);
+
+    constructor(
+        address admin,
+        address _kyberNetwork,
+        uint16 _rewardBps, 
+        uint16 _rebateBps, 
+        uint16 _burnBps
+    ) public PermissionGroupsNoModifiers(admin) {
+        require(_burnBps + _rewardBps + _rebateBps == BPS, "Bad BRR values");
```

Comment:
can use safe math and remove extra requires.

we did same change in our Katalyst.

Reply:
change as request